gluster-exporter/metrics.go

```diff
@@ -29,12 +33,26 @@ func (m *Metric) LabelNames() []string {
 	return out
 }
 
-// newPrometheusGaugeVec is wrapper around prometheus.NewGaugeVec. Additionally
-// it registers with global map which can be used for documentation,
-// listing supported metrics via CLI etc.
-// 'gaugeVecArrPtr' helps to collect all the 'GaugeVec' pertaining to a metric
-// group.  Ignored if 'nil'
-func newPrometheusGaugeVec(m Metric, gaugeVecArrPtr *[]*prometheus.GaugeVec) *prometheus.GaugeVec {
+var metrics []Metric
+var defaultMetricTTL = 2 * time.Minute
+
+type MetricWithTTL struct {
+	LastUpdated time.Time
+	Labels      prometheus.Labels
+}
+
+type ExportedGaugeVec struct {
+	Namespace string
+	Name      string
+	Help      string
+	LongHelp  string
+	Labels    []string
+	GaugeVec  *prometheus.GaugeVec
+	Metrics   map[uint64]MetricWithTTL
```

Comment:
A bit confused as to why a map is used here. I see in https://github.com/gluster/gluster-prometheus/pull/159/files#diff-ea1e4a81e69f73530ae83d5e047756b3R91 , a hash is computed but unclear as to what is the purpose of using the hash(and the map) since we are dealing with a single unique metric here.

Reply:
One GaugeVec is a collection of metrics with different label combinations. For example, `bricks_count` is a GaugeVec with `brick_count{volname=gv1}` and `brick_count{volname=gv2}` and other metrics.

Reply:
Ahh got it! But do we have to compute the hash each time? Since we are iterating all the metrics in the GaugeVec in RemoveStaleMetrics , can't we using use an array of metrics instead?

Reply:
Ignore the previous comment... I see that in SetMetricLastUpdated we need to compute the hash for that particular label/metric and set the TTL value and hash of label would be the way to distinguish

Reply:
hash is also not expensive, just a sorted list of labels https://github.com/prometheus/common/blob/master/model/signature.go#L33

Reply:
Ack. Thanks!